spec/mixins/webhook_spec.rb

```diff
@@ -1,29 +1,39 @@
 require "spec_helper"
 
-RSpec.describe Rubyblok::Mixins::Webhook, type: :controller do
-  describe "#create" do
-
-    let(:subject) { Class.new { include Rubyblok::Mixins::Webhook } }
-    let(:instance) { subject.new }
+class Page
+end
 
+RSpec.describe WebhookController do
+  describe "#create" do
+    let(:subject) { WebhookController.new }
     let(:api_token)      { "api_token" }
     let(:version)        { "version" }
     let(:component_path) { "shared" }
     let(:webhook_secret) { "webhook_secret" }
-    let(:params) { { text: "text", action: "publish", space_id: 123, story_id: 456 }.to_json }
+    let(:model_name) { "Page" }
+    let(:params) { { "text" => "text", "action" => "publish", "space_id" => 123, "story_id" => 456, "name" => "home" } }
+    let(:page) { Page.new }
 
     before do
-      allow(instance).to receive(:render).with(json: { success: true }).and_return({ success: true })
-      allow(Rubyblok.configuration).to receive_messages(api_token: api_token,
-                                                        version: version,
-                                                        component_path: component_path,
-                                                        webhook_secret: webhook_secret)
+      allow(subject).to receive_message_chain(:request, :raw_post).and_return(params.to_json)
+      allow(subject).to receive_message_chain(:storyblok_story_content).and_return(params)
+      allow(Rubyblok.configuration).to receive_messages(api_token:,
+                                                        version:,
+                                                        component_path:,
+                                                        webhook_secret:,
+                                                        model_name:)
+
+      allow(Rubyblok::Services::GetStoryblokStory).to receive(:call).and_return(params)
+
+      allow(Page).to receive(:find_or_initialize_by).and_return(page)
+      allow(page).to receive(:update_attribute)
     end
 
     context "when the webhook receives an update" do
       it "renders the correct response" do
-        expect(instance).to respond_to(:create)
-        expect(instance.create).to eq({ success: true })
+        expect(subject).to receive(:render).with(json: { success: true })
+        expect(page).to receive(:update).with(storyblok_story_content: params, storyblok_story_slug: "home")
```

Comment:
We should not use mutiple expects in the same `it` block, it should be one `expect` per block.